File: src/dorkbox/network/connection/registration/UpgradeType.java
package dorkbox.network.connection.registration;

public
class UpgradeType {
    // The check is > 0, so these MUST be all > 0

    public static final byte NONE = (byte) 1;
    public static final byte ENCRYPT = (byte) 2;
    public static final byte COMPRESS = (byte) 3;
    public static final byte FRAGMENTED = (byte) 4;
}

File: src/dorkbox/network/handshake/UpgradeType.kt
/*
 * Copyright 2020 dorkbox, llc
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package dorkbox.network.handshake

object UpgradeType {
    // The check is > 0, so these MUST be all > 0
    const val NONE = 1.toByte()
    const val ENCRYPT = 2.toByte()
    const val COMPRESS = 3.toByte()
    const val FRAGMENTED = 4.toByte()
}
